datatypes/src/raster/operations/interpolation.rs

```diff
@@ -126,58 +109,53 @@ where
         let out_x_size = info_out.global_geo_transform.x_pixel_size();
         let out_y_size = info_out.global_geo_transform.y_pixel_size();
 
-        let parallelism = rayon::current_num_threads();
-        let rows_per_task =
-            num::integer::div_ceil(output.grid_array.shape.axis_size_y(), parallelism);
+        let map_fn = |g_idx: GridIdx2D, _current_pixel_value: Option<P>| {
```

Comment:
since wa cant have a "&mut pixel" because the pixel is currently a value and a bool we need to pass the current value in. It is a copy type so there should be no harm. The funtion then updates the value with the result value.

Reply:
couldn't we just have to types of map iterations? One that takes index and pixel value and one that gives only the index so we don't carry over unneeded data?